homeassistant/helpers/entity_component.py

```diff
@@ -229,9 +229,23 @@ async def async_setup_platform(
             return
 
         # Use config scan interval, fallback to platform if none set
-        scan_interval = platform_config.get(
-            CONF_SCAN_INTERVAL, getattr(platform, "SCAN_INTERVAL", None)
-        )
+        # FIXME: Components don't pass discovery_info in the correct format:
+        # - Updater passes an empty dictionary instead of None
+        # - Cover passes a list containing a single dictionary when initialized
+        # etc.mm@PentaphyllumYubiKey
+        if isinstance(discovery_info, list):
+            scan_interval = discovery_info[0].get(
```

Comment:
Integrations are responsible for the discovery info. It's not standardized and we don't support setting the scan interval via the discovery info.

Please remove these changes.

Reply:
OK, but how I suppose to make a 1-to-1 transition into a new config? Scan interval is set under the platform like this:
```yaml
cover:
 - platform: modbus
   scan_interval: 10
```

This attribute is essential for my component, and as you see it's handled outside of my code. In the `cover` platform itself.

Reply:
Why is the option essential?

The common scan interval option is legacy. If an integration needs to adjust the scan interval dynamically or heterogenously it has to take control of polling either via our coordinator helper or via a custom update logic using our event helper.

Reply:
By default, `cover` polls for updates once in 15 seconds. In my case, I need to read data once in a second.
If the `scan_interval` is legacy, maybe it should be mentioned in the documentation.

Anyway, if there's another way how to control this, I can rework the code.

So basically, I will set my own attribute `scan_interval` and use your helper methods to pass it on? Could you point me to some examples? For starters, please let me know which helpers did you mean, so that I can look at the code.

Thanks!

Reply:
I found some other integrations with the new approach. I'll try to modify my code and will notify you...

Reply:
The lowest allowed scan interval is 5 seconds.

If all platform instances can have the same scan interval we can just define the `SCAN_INTERVAL` constant in the platform module and it will override the default scan interval.